app/uk/gov/hmrc/gform/builder/BuilderController.scala

```diff
@@ -632,4 +633,147 @@ class BuilderController(
+    ) { implicit request => lang => cache => sse => formModelOptics =>
+      gformConnector.getFormTemplateRaw(formTemplateId).flatMap { formTemplateRaw =>
+        val jsonString: String = PlayJson.stringify(formTemplateRaw)
+        val maybeCirceJson: Either[ParsingFailure, Json] =
+          io.circe.parser.parse(jsonString)
+
+        val json: Option[Json] = maybeCirceJson.toOption.flatMap { json =>
+          json.hcursor.downField("acknowledgementSection").focus
+        }
+        json match {
+          case None                         => InternalServerError("Unable to retrieve acknowledgement section").pure[Future]
+          case Some(acknowledgementSection) => Ok(acknowledgementSection).pure[Future]
+        }
+      }
+
+    }
+
+  def generateAcknowledgementPanelHtml(formTemplateId: FormTemplateId) =
+    auth.authAndRetrieveForm[SectionSelectorType.Normal](
+      formTemplateId,
+      None,
+      OperationWithForm.EditFormAcknowledgement
+    ) { implicit request => _ => cache => implicit sse => formModelOptics =>
+      import i18nSupport._
+      val formTemplate = cache.formTemplate
+      val envelopeId = cache.form.envelopeId
+
+      val destinationList = formTemplate.destinations match {
+        case destinationList: DestinationList => destinationList
+        case _                                => throw new Exception("")
+      }
+
+      val formCategory = formTemplate.formCategory
+      val panelTitle = destinationList.acknowledgementSection.panelTitle.map(_.value())
+      val showReference = destinationList.acknowledgementSection.showReference
+      val heading = renderer.acknowledgementHeading(formCategory)
+
+      val html = renderer.renderAcknowledgementPanel(
+        panelTitle,
+        showReference,
+        formCategory,
+        envelopeId,
+        heading
+      )
+      Ok(
+        Json.obj(
+          "panelHtml" := sanitiseHtml(html)
+        )
+      ).pure[Future]
+    }
+
+  def htmlForAcknowledgementFormComponent(
+    formTemplateId: FormTemplateId,
+    formComponentId: FormComponentId
+  ) =
+    auth.authAndRetrieveForm[SectionSelectorType.Normal](
+      formTemplateId,
+      None,
+      OperationWithForm.EditFormAcknowledgement
+    ) { implicit request => implicit lang => cache => implicit sse => formModelOptics =>
+      import i18nSupport._
+
+      val maybeAcknowledgementSection = cache.formTemplate.destinations.fold[Option[AcknowledgementSection]](ds =>
+        Option[AcknowledgementSection](ds.acknowledgementSection)
+      )(_ => Option.empty[AcknowledgementSection])
+      maybeAcknowledgementSection match {
+        case Some(acknowledgementSection) =>
+          renderHtmlForAcknowledgementFormComponent(
+            acknowledgementSection,
+            formComponentId,
+            formModelOptics,
+            cache
+          )
+        case _ =>
+          BadRequest("Can not find acknowledgement section in the form").pure[Future]
```

Comment:
As above this needs to use `badRequest` method

Reply:
done